src/snapred/ui/widget/MultiSelectDropDown.py

```diff
@@ -0,0 +1,74 @@
+from qtpy.QtCore import Qt
+from qtpy.QtGui import QStandardItemModel
+from qtpy.QtWidgets import QComboBox, QStyledItemDelegate, QVBoxLayout, QWidget
+
+
+class CheckableComboBox(QComboBox):
+    def __init__(self, parent=None):
+        super(CheckableComboBox, self).__init__(parent)
+        self.view().pressed.connect(self.handleItemPressed)
+        self.setModel(QStandardItemModel(self))
+
+    def handleItemPressed(self, index):
+        item = self.model().itemFromIndex(index)
+        if item.checkState() == Qt.Checked:
+            item.setCheckState(Qt.Unchecked)
+        else:
+            item.setCheckState(Qt.Checked)
+
+    def checkedItems(self):
```

Comment:
Please do a pass and `annotate` a few of these methods, or at _least_ this one.  For example, I think this is `checkedItems(self): -> List[QString]` which really helps see what's precisely going on.  (No need to annotate _everything_, but return values help.  :)  )

Reply:
Done!